hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java

```diff
@@ -246,6 +254,28 @@ UsageAccumulator baseUsageFor(HederaFunctionality function, SubType type) {
 		throw new IllegalArgumentException("Canonical usage unknown");
 	}
 
+	UsageAccumulator contractAutoRenew() {
+		final var accountContext = ExtantCryptoContext.newBuilder()
+				.setCurrentExpiry(0)
+				.setCurrentMemo(BLANK_MEMO)
+				.setCurrentKey(A_KEY)
+				.setCurrentlyHasProxy(false)
+				.setCurrentNumTokenRels(0)
+				.setCurrentMaxAutomaticAssociations(0)
+				.setCurrentCryptoAllowances(Collections.emptyList())
+				.setCurrentTokenAllowances(Collections.emptyList())
+				.setCurrentApproveForAllNftAllowances(Collections.emptyList())
+				.build();
+		final var contractContext = new ExtantContractContext(
+				CANONICAL_NUM_CONTRACT_KV_PAIRS,
+				CANONICAL_CONTRACT_BYTECODE_SIZE,
+				accountContext);
+		final var into = new UsageAccumulator();
```

Comment:
Not in this case, because a `ContractAutoRenew` isn't a true user transaction that requires precheck, gossip, etc.; so none of the `vpt`/`bpt`/`bpr`/`sbpr` resources are used.